### PR TITLE
test: Migrate git to snapbox

### PR DESCRIPTION
**Pull request by dieterplex**
Part of #14039.

There is a case need to modify regex for file size redaction.

### CODE REVIEW

File: tests/testsuite/git.rs [hunk 1]
```diff
@@ -2821,23 +2840,24 @@ fn use_the_cli() {
...
-[LOCKING] [..]
-[CHECKING] dep1 [..]
-[RUNNING] `rustc [..]`
-[CHECKING] foo [..]
-[RUNNING] `rustc [..]`
-[FINISHED] [..]
-";
+    let stderr = str![[r#"
+[UPDATING] git repository `[ROOTURL]/dep1`
+[RUNNING] `git fetch --verbose --force --update-head-ok [..][ROOTURL]/dep1[..] [..]+HEAD:refs/remotes/origin/HEAD[..]]`
```
Comment (dieterplex) [inline tests/testsuite/git.rs:2845]: There is a quote difference between Win and Linux. Not sure if it is an issue.

```
   2      - [RUNNING] `git fetch --verbose --force --update-head-ok '[ROOTURL]/dep1' '+HEAD:refs/remotes/origin/HEAD'`
        2 + [RUNNING] `git fetch --verbose --force --update-head-ok [ROOTURL]/dep1 +HEAD:refs/remotes/origin/HEAD`
```

File: tests/testsuite/git.rs [hunk 2]
```diff
@@ -550,10 +556,10 @@ fn cargo_compile_with_short_ssh_git() {
 
     p.cargo("check")
         .with_status(101)
-        .with_stdout("")
-        .with_stderr(&format!(
+        .with_stdout_data(str![])
```
Comment (weihanglo) [inline tests/testsuite/git.rs:559]: This is likely not needed.
```suggestion
```

File: tests/testsuite/git.rs [hunk 3]
```diff
@@ -58,26 +57,23 @@ fn cargo_compile_simple_git_dep() {
...
+
+"#]])
         .run();
 
     assert!(project.bin("foo").is_file());
 
     project
         .process(&project.bin("foo"))
-        .with_stdout("hello world\n")
+        .with_stdout_data("hello world\n")
```
Comment (weihanglo) [inline tests/testsuite/git.rs:76]: nit: could wrap with `str![]`.

File: tests/testsuite/git.rs [hunk 4]
```diff
@@ -2770,14 +2779,24 @@ fn failed_submodule_checkout() {
...
         .with_status(101)
-        .with_stderr_contains("  failed to update submodule `src`")
-        .with_stderr_contains("  failed to update submodule `bar`")
+        .with_stderr_data(str![[r#"
+...
+  failed to update submodule `src`
+...
+  failed to update submodule `bar`
+...
+"#]])
```
[the remark this answers is not in the document]
Comment (dieterplex) [reply tests/testsuite/git.rs:2799]: executing `__CARGO_USE_GITOXIDE_INSTEAD_OF_GIT2=1 SNAPSHOTS=overwrite cargo test git::failed_submodule_checkout` would given below diff:
```
---- expected: tests/testsuite/git.rs:2816:27
++++ actual:   stderr
   1    1 | [UPDATING] git repository `[ROOTURL]/dep1`
   2      - [UPDATING] git submodule `https://127.0.0.1:33953/`
        2 + [UPDATING] git submodule `https://127.0.0.1:45123/`
        3 + [WARNING] spurious network error (3 tries remaining): [35] SSL connect error (OpenSSL/3.3.1: [ERROR]0A00010B:SSL routines::wrong version number)
        4 + [WARNING] spurious network error (2 tries remaining): [35] SSL connect error (OpenSSL/3.3.1: [ERROR]0A00010B:SSL routines::wrong version number)
        5 + [WARNING] spurious network error (1 tries remaining): [35] SSL connect error (OpenSSL/3.3.1: [ERROR]0A00010B:SSL routines::wrong version number)
   3    6 | [ERROR] failed to get `dep1` as a dependency of package `foo v0.5.0 ([ROOT]/foo)`
   4    7 | 
   5    8 | Caused by:
   6    9 |   failed to load source for dependency `dep1`
   7   10 | 
          ⋮
  13   16 | 
  14   17 | Caused by:
  15   18 |   failed to update submodule `bar`
  16   19 | 
  17   20 | Caused by:
  18      -   failed to fetch submodule `bar` from https://127.0.0.1:33953/
       21 +   failed to fetch submodule `bar` from https://127.0.0.1:45123/
  19   22 | 
  20   23 | Caused by:
  21   24 |   network failure seems to have happened
  22   25 |   if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  23   26 |   https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
  24   27 | 
  25   28 | Caused by:
  26      -   SSL [ERROR] [ERROR]0A00010B:SSL routines::wrong version number; class=Ssl (16)
       29 +   An IO error occurred when talking to the server
       30 + 
       31 + Caused by:
       32 +   [35] SSL connect error (OpenSSL/3.3.1: [ERROR]0A00010B:SSL routines::wrong version number)
```

`git_auth::https_something_happens` is also handling different error messages on different platforms and the address may differ each time.
Should we focus on things important? snapshot may not work well with address, or redact it?

[the remark this answers is not in the document]
Comment (weihanglo) [reply tests/testsuite/git.rs:2810]: Fair enough. Let us keep your change.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Looks great. Thanks!

@bors r+

**Pull request merged**